src/ports/postgres/modules/deep_learning/madlib_keras_fit_multiple_model.py_in

```diff
@@ -337,183 +376,308 @@ class FitMultipleModel():
-        for mst, dist_key in zip(mst_row, self.dist_keys):
-            if mst:
-                model_id = mst[self.model_id_col]
-                compile_params = mst[self.compile_params_col]
-                fit_params = mst[self.fit_params_col]
-                mst_key = mst[self.mst_key_col]
-                object_map = mst[self.object_map_col]
-            else:
-                model_id = "NULL"
-                compile_params = "NULL"
-                fit_params = "NULL"
-                mst_key = "NULL"
-                object_map = None
-            mst_insert_query = plpy.prepare(
-                               """
-                               INSERT INTO {self.mst_current_schedule_tbl}
-                                   VALUES ({model_id},
-                                           $madlib${compile_params}$madlib$,
-                                           $madlib${fit_params}$madlib$,
-                                           {dist_key},
-                                           {mst_key},
-                                           $1)
-                                """.format(**locals()), ["BYTEA"])
-            plpy.execute(mst_insert_query, [object_map])
-
-    def create_model_output_table(self):
-        output_table_create_query = """
-                                    CREATE TABLE {self.model_output_table}
-                                    ({self.mst_key_col} INTEGER PRIMARY KEY,
-                                     {self.model_weights_col} BYTEA,
-                                     {self.model_arch_col} JSON)
-                                    """.format(self=self)
-        plpy.execute(output_table_create_query)
-        self.initialize_model_output_and_info()
+                custom_fn_names.add(local_metric)
+                custom_msts.append(mst)
+
+        self.custom_fn_object_map = query_custom_functions_map(self.object_table, custom_fn_names)
+
+        for mst in custom_msts:
+            mst[self.object_map_col] = self.custom_fn_object_map
+
+        self.custom_mst_keys = { mst['mst_key'] for mst in custom_msts }
+
+    def init_schedule_tbl(self):
+        self.prev_dist_key_col = '__prev_dist_key__'
+        mst_key_list = '[' + ','.join(self.all_mst_keys) + ']'
+
+        create_sched_query = """
+            CREATE TABLE {self.schedule_tbl} AS
+                WITH map AS
+                    (SELECT
+                        unnest(ARRAY{mst_key_list}) {self.mst_key_col},
+                        unnest(ARRAY{self.all_dist_keys}) {self.dist_key_col}
+                    )
+                SELECT
+                    map.{self.mst_key_col},
+                    {self.model_id_col},
+                    map.{self.dist_key_col} AS {self.prev_dist_key_col},
+                    map.{self.dist_key_col}
+                FROM map LEFT JOIN {self.model_selection_table}
+                    USING ({self.mst_key_col})
+            DISTRIBUTED BY ({self.dist_key_col})
```

Comment:
We would if we wanted the weights to stay fixed... but the purpose of this query is to redistribute the weights from the previous distribution to the next distribution.  The output has to be distributed by __dist_key__ since that's what will be used as the JOIN key with the source table's __dist_key__ in the next query.

Reply:
Oh, I responded at first thinking this was the hop query.

The distribution of this table shouldn't matter much, since there are no weights in it... but yes, I think you're right, I'll change it to prev dist key.

Reply:
Actually, I decided I'm going to leave it `dist_key_col` here, but add a distribution rule for the rotate_schedule_tbl query to DISTRIBUTE BY `prev_dist_key`, as you suggested in another comment.

The first hop is always just a table rename, so the initial schedule table never actually gets used for the hop itself.  It's only used to generate the next schedule table in the rotate_schedule_tbl query.  And in that query, it's the `dist_key_col` that matters... `prev_dist_key_col` is unused, I guess we really don't need that column to exist at all.. it's just there as a placeholder so that the schema is the same as the next schedule table that gets generated later by `rotate_schedule_tbl()`.